babel/messages/_compat.py

```diff
@@ -0,0 +1,32 @@
+from functools import partial
+
+
+def find_entrypoints(group_name: str):
+    """
+    Find entrypoints of a given group using either `importlib.metadata` or the
+    older `pkg_resources` mechanism.
+
+    Yields tuples of the entrypoint name and a callable function that will
+    load the actual entrypoint.
+    """
+    try:
+        from importlib.metadata import entry_points
+    except ImportError:
+        pass
+    else:
+        eps = entry_points()
+        if isinstance(eps, dict):  # Old structure before Python 3.10
+            group_eps = eps.get(group_name, [])
+        else:  # New structure in Python 3.10+
+            group_eps = (ep for ep in eps if ep.group == group_name)
+        for entry_point in group_eps:
+            yield (entry_point.name, entry_point.load)
+        return
+
+    try:
+        from pkg_resources import working_set
+    except ImportError:
+        pass
+    else:
+        for entry_point in working_set.iter_entry_points(group_name):
+            yield (entry_point.name, partial(entry_point.load, require=True))
```

Comment:
As my colleague @j123b567 rightly noted, for versions before 3.10, it is better to use `pkg_resources`, as the `importlib.metadata` API is still unstable. 
You can use `pkg_resources` as the first handler, and in the second try block, check that the object type is not a dictionary before using `importlib.metadata`. Otherwise, simply continue the function execution. 
Consider this option as well.

Reply:
Could you post this as a code suggestion so it's easier to see what you mean?

Reply:
> Could you post this as a code suggestion so it's easier to see what you mean?

yeah

Reply:
```suggestion
    try:
        from pkg_resources import working_set
    except ImportError:
        pass
    else:
        for entry_point in working_set.iter_entry_points(group_name):
            yield (entry_point.name, partial(entry_point.load, require=True))
        return
    try:
        from importlib.metadata import entry_points
    except ImportError:
        pass
    else:
        eps = entry_points()
        if not isinstance(eps, dict):  # New structure in Python 3.10+
            group_eps = (ep for ep in eps if ep.group == group_name)
            for entry_point in group_eps:
                yield (entry_point.name, entry_point.load)
```

Reply:
Good idea – I pushed a commit with a variation on this idea, namely to only even try importlib.metadata on Python 3.10+, when we know it will be stable.

Reply:
In my initial local implementation, there was a strict check for the Python version, but we decided to abandon this approach since various Python interpreters (e.g., PyPy) and, consequently, different standard modules can be used. My proposed version is universal and not tied to a specific version, but only to the availability of the library.